### ISSUE TITLE
``make latexpdf`` after ``make latex`` rebuilds latex sources

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#4362]
#### Procedure to reproduce the problem

With any project, ``make latexpdf`` will unneedlessly rebuild the latex source files even immediately afer ``make latex``.

This is noticeable on big enough projects such as Sphinx own docs.

#### Expected results

``make latexpdf`` should behave like ``make -C <builddir> all-pdf`` when there is no need to rebuild the latex files.

### Environment info
- OS: Mac
- Python version: 3.6.4
- Sphinx version: 1.6.5

May be relevant to #4343

#### edit

This actually applies to ``make latex`` itself: if executed twice in a row with no intervening changes, it still re-creates the tex file (which takes time on big projects).

**Comment by jfbu** [comment]
For example when building Sphinx documentation at my locale, I currently get typically:

```
The LaTeX files are in _build/latex.
Run 'make' in that directory to run these through (pdf)latex
(use `make latexpdf' here to do that automatically).

real	0m45.885s
user	0m45.703s
sys	0m0.172s
```

And if I now do

```
$ time make -C _build/latex all-pdf
```

this gives

```
real	0m9.943s
user	0m9.660s
sys	0m0.258s
```

This is why it is frustrating that `make latexpdf` immediately following `make latex` redoes all the Sphinx phase which amounts for most of the build time.

**Comment by tk0miya** [comment]
Fixed by #5085.
Please let me know if it does not work well.
